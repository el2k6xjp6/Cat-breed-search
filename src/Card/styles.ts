import styled from "styled-components";

export const Container = styled.div`
  width: calc(33% - 22px);
  margin: 10px 10px;
  height: 400px;
  border-radius: 24px;
  border: 1px solid #c4c4c4;
`;

export const Image = styled.img`
  width: 100%;
  height: 300px;
  object-fit: cover;
  border-top-left-radius: 24px;
  border-top-right-radius: 24px;
`;

const Text = styled.div`
  text-indent: 15px;
  height: 30px;
  line-height: 30px;
`;

export const Name = styled(Text)`
  font-size: 18px;
`;

export const LifeSpan = styled(Text)`
  font-size: 14px;
`;

export const Weight = styled(Text)`
  font-size: 14px;
`;
