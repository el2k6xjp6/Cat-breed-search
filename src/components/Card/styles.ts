import styled from "styled-components";

export const Container = styled.div`
  width: 300px;
  margin: 10px 10px;
  height: 400px;
  border-radius: 24px;
  border: 1px solid #c4c4c4;
`;

const Text = styled.div`
  text-indent: 15px;
  height: 30px;
  line-height: 30px;
`;

export const Name = styled(Text)`
  font-size: 18px;
`;

export const LifeSpan = styled(Text)`
  font-size: 14px;
`;

export const Weight = styled(Text)`
  font-size: 14px;
`;
