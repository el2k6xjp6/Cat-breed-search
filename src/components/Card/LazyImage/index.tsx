import React from "react";
import { StyledImage } from "./styles";

interface Props {
  isFetching: boolean;
  actualSrc: string;
}
function LazyImage({ actualSrc, isFetching }: Props) {
  const [isImageLoaded, setImageLoaded] = React.useState(false);
  // const [hasError, setHasError] = React.useState(false);

  React.useEffect(() => {
    const img = new Image();

    img.onload = () => setImageLoaded(true);
    // img.onerror = () => setHasError(true);

    img.src = actualSrc;
  }, [actualSrc]);

  return (
    <StyledImage
      actualSrc={actualSrc}
      loading={!isImageLoaded || isFetching}
      // error={hasError}
      // {...props}
    />
  );
}

export default LazyImage;
