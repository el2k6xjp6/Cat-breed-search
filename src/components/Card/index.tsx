import React, { useEffect, useState } from "react";
import { Container, Name, LifeSpan, Weight } from "./styles";
import LazyImage from "./LazyImage";
import { fetchTheCatImageById } from "../../utils";

interface CardType {
  name: string;
  weight: { metric: string };
  life_span: string;
  reference_image_id: string;
}

function Card(props: CardType) {
  const [isFetching, setFetching] = React.useState<boolean>(true);

  useEffect(() => {
    (async () => {
      const data = await fetchTheCatImageById(props.reference_image_id);
      setImageUrl(data.url);
      setFetching(false);
    })();
  }, [props.reference_image_id]);

  const [imageUrl, setImageUrl] = useState<string>("");

  return (
    <Container>
      <LazyImage isFetching={isFetching} actualSrc={imageUrl} />
      <Name>{props.name}</Name>
      <LifeSpan>Lifespan: {props.weight.metric} kilograms</LifeSpan>
      <Weight>Weight: {props.life_span} years</Weight>
    </Container>
  );
}

export default React.memo(Card);
