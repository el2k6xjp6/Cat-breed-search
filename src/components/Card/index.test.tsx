import Card from ".";
import "jest-styled-components";
import renderer from "react-test-renderer";

const mockProps = {
  name: "test",
  weight: { metric: "1-2" },
  life_span: "3-4",
  reference_image_id: "",
};

describe("App", () => {
  test("snapshot", () => {
    const tree = renderer.create(<Card {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
