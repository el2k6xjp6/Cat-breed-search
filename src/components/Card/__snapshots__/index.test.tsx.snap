// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
.c0 {
  width: 300px;
  margin: 10px 10px;
  height: 400px;
  border-radius: 24px;
  border: 1px solid #c4c4c4;
}

.c2 {
  text-indent: 15px;
  height: 30px;
  line-height: 30px;
}

.c3 {
  font-size: 18px;
}

.c4 {
  font-size: 14px;
}

.c5 {
  font-size: 14px;
}

.c1 {
  background-image: url("Spinner.gif");
  background-position: center center;
  background-size: cover;
  width: 100%;
  height: 300px;
  object-fit: cover;
  border-top-left-radius: 24px;
  border-top-right-radius: 24px;
}

<div
  className="c0"
>
  <div
    className="c1"
    loading={true}
  />
  <div
    className="c2 c3"
  >
    test
  </div>
  <div
    className="c2 c4"
  >
    Lifespan: 
    1-2
     kilograms
  </div>
  <div
    className="c2 c5"
  >
    Weight: 
    3-4
     years
  </div>
</div>
`;
